### PR TITLE
[Frontend] Add Holo2 reasoning parser

### PR DESCRIPTION
**Pull request by secbear**
## Purpose

Add Holo2 reasoning-parser support to the Rust frontend parity roadmap.

Python registers `holo2` through `DeepSeekV3ReasoningWithThinkingParser`, which defaults to DeepSeek R1 behavior. The Rust `DeepSeekV3ReasoningParser` alias has a different no-boundary default, so this adds a semantic `Holo2ReasoningParser` alias directly to the prompt-first `DeepSeekR1ReasoningParser`.

The change:

- exposes `holo2` as an explicit reasoning-parser name;
- automatically routes `HCompany/Holo2-*` models to it;
- reuses the shared `<think>...</think>` parser without adding another state machine;
- adds factory registration and model-routing coverage;
- adds a real `HCompany/Holo2-4B` tokenizer/template roundtrip covering enabled, disabled, and default thinking behavior.

No tool-parser or inference-engine behavior is changed.

Refs #44280

### Duplicate-work check

I reviewed the roadmap comments and searched open PRs by the roadmap issue number and Holo2/Rust reasoning-parser keywords. No existing claim or open PR overlaps this implementation.

## Test Plan

- Verify explicit parser registration and factory listing.
- Verify automatic routing of `HCompany/Holo2-4B` to `holo2`.
- Exercise the real Holo2 tokenizer and chat template through the existing reasoning-and-content roundtrip.
- Cover `thinking=true`, `thinking=false`, and omitted/default thinking.
- Run the complete `vllm-parser` and `vllm-chat` test suites.
- Run repository pre-commit hooks on the changed files.

## Test Result

Passed:

```text
cargo fmt --manifest-path rust/Cargo.toml --all --check
```

Passed all 698 tests:
```text
nix develop path:. cargo nextest run \
  --manifest-path rust/Cargo.toml \
  --locked \
  -p vllm-parser \
  -p vllm-chat
```

Passed:
```text
nix develop path:. -c pre-commit run
```

The real HCompany/Holo2-4B tokenizer/template roundtrip passed for
enabled, disabled, and default thinking.

Model-generation or accuracy evaluation is not applicable because this change does not modify inference, logits, or model output generation. The real-tokenizer/template roundtrip provides parser integration evaluation.
Documentation was not changed because the existing reasoning-output documentation already lists holo2; this PR adds Rust frontend parity.

## AI assistance disclosure
AI assistance was used for repository exploration, implementation planning, code review, and identifying a model-ID casing correction. I reviewed every changed line and ran all tests listed above.

### CODE REVIEW

**Comment by BugenZhao** [comment]
@codex review
